src/client/dfuse/test/test_ioil.c

```diff
@@ -43,8 +43,8 @@
  * warning in sanity(...) with some newer compilers
  */
 static char *mount_dir = "/tmp";
-static uint64_t max_read_size;
-static uint64_t max_iov_read_size;
+static uint64_t max_read_size = 4096;
```

Comment:
Was this intentional?  I think they were supposed to be set to zero, however this is also test code that we don't execute currently so I'm not sure.

Reply:
The bug that was detected was that the new flags were initializing them to 0 which caused an error in a calculation.  I asked about this in the dev channel and got no response. This was my original question


" I'm trying to enable compiler hardening in the daos build and this one test is throwing a sanity check failure for obvious reasons but I'd like some input on selecting values to fix it

src/client/dfuse/test/test_ioil.c: line 230 says its a 0 size memset which it is. buf_size is based on test3_size which is test2_size + max_iov_read_size, which is test1_size + max_iov_read_size which is max_read_size * 2.

So its (max_read_size * 2) + max_iov_read_size + max_iov_read_size. The problem is max_read_size and max_iov_read_size are both declared static up top in the file and not assigned a value so the compiler is making them 0. So its 0 * 2 + 0 + 0 which is where the error comes from. Do we know what these values should be set to? "

Reply:
@ashleypittman they are values that were read from ctrlfs in IOF in init_suite.   As there is no ctrlfs anymore, I presume they would need to be read from elsewhere or initialized to some value.   Seems 4096 is as good as any default to get rid of the compiler error in the interim.